contracts/cw1-subkeys/src/contract.rs

```diff
@@ -196,6 +236,47 @@ where
     Ok(res)
 }
 
+pub fn handle_setup_permissions<S: Storage, A: Api, Q: Querier, T>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    spender: HumanAddr,
+    permissions: Permissions,
+) -> StdResult<HandleResponse<T>>
+    where
+        T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    let cfg = admin_list_read(&deps.storage).load()?;
+    let spender_raw = &deps.api.canonical_address(&spender)?;
+    let owner_raw = &deps.api.canonical_address(&env.message.sender)?;
+
+    if !cfg.is_admin(&owner_raw) {
+        return Err(StdError::unauthorized());
+    }
+    if spender_raw == owner_raw {
+        return Err(StdError::generic_err("Cannot set allowance to own account"));
+    }
+
+    // update allowance permissions
+    allowances(&mut deps.storage).update(spender_raw.as_slice(), |allow| {
```

Comment:
We should be able to set the staking permissions on accounts without any balances. 

It may just have balances, or just permissions, or both (and they should not interfere with each other)

Reply:
On it